server/routes/customer.js

```diff
@@ -5,37 +5,56 @@ import dbConfig from '../dbconfig.js';
 const router = express.Router();
 
 router.get('/', function (req, res) {
-  const authenticatedUserId = req.session.authenticatedUser;
-  const getData = async () => {
-    try {
-      let pool = await sql.connect(dbConfig);
-
-      let orderResults = await pool
-        .request()
-        .input('customerId', sql.Int, authenticatedUserId)
-        .query(
-          'SELECT customerId, firstName, lastName, email, phonenum, address, city, state, postalCode, country, userid FROM customer WHERE customerId = @customerId'
-        );
-
-      res.write(JSON.stringify(orderResults));
-      res.end();
-    } catch {
-      (error) => {
-        console.log(error);
-        res.write(error);
-        console.dir(err);
-
-        res.end();
-      };
-    }
-  };
-
-  if (authenticatedUserId) {
-    getData();
-  } else {
-    res.write('unauthenticated');
+  if (!req.session.authenticatedUser) {
     res.end();
+    return;
   }
+
+  getCustomer(req.session.authenticatedUser)
+    .then((customer) => {
+      // undefined if no customer found
+      res.send(customer);
+    })
+    .catch(() => {
+      res.end();
+    });
 });
 
+function getCustomer(customerId) {
+  return new Promise((resolve, reject) => {
+    sql.connect(dbConfig).then((conn) => {
+      conn
+        .request()
+        .input('customerId', sql.Int, customerId)
+        .query(
+          `SELECT 
+            customerId AS id,
+            firstName,
+            lastName,
+            email,
+            phonenum AS phone,
+            address,
+            city,
+            state,
+            postalCode,
+            country,
+            userid AS username,
+            password
```

Comment:
I'm not sure we should be including the password in this query, since the whole result is then sent to the front end, and also since we don't use it (I think).